core/server/api/shared/pipeline.js

```diff
@@ -0,0 +1,142 @@
+const debug = require('ghost-ignition').debug('api:shared');
+const Promise = require('bluebird');
+const _ = require('lodash');
+const shared = require('../shared');
+const common = require('../../lib/common');
+const sequence = require('../../lib/promise/sequence');
+
+const STAGES = {
+    validation: {
+        input(apiUtils, apiConfig, apiImpl, options) {
+            debug('stages: validation');
+            const ops = [];
+
+            // CASE: do validation completely yourself
+            if (typeof apiImpl.validation === 'function') {
+                return apiImpl.validation(options);
+            }
+
+            ops.push(function doValidation() {
+                return shared.validators.handle.input(
+                    Object.assign({}, apiConfig, apiImpl.validation),
+                    apiUtils.validators.input,
+                    options
+                );
+            });
+
+            return sequence(ops);
```

Comment:
I had some more tasks in it, because i was supporting `before` and `after` hooks, but as we don't use them now, i just reduced my work. This logic just offers the ability to extend the validation stage.